message/history/eventFinder.go

```diff
@@ -23,90 +40,132 @@ func (f FinderFunc) Find(events []message.Event, currentEvent message.Event) (me
 // with the boot-time of the previous session. If any of the fatalValidators returns false,
 // it will stop searching and immediately exit, returning the event
 // that prompted the error along with the error returned by the fatalValidator.
-func LastSessionFinder(validators validation.Validator, fatalValidators validation.Validator) FinderFunc {
+func LastSessionFinder(validator validation.Validator, fatalValidator validation.Validator) FinderFunc {
 	return func(events []message.Event, currentEvent message.Event) (message.Event, error) {
 		// verify that the current event has a boot-time
 		currentBootTime, err := currentEvent.BootTime()
 		if currentBootTime <= 0 {
 			return message.Event{}, validation.InvalidBootTimeErr{OriginalErr: err}
 		}
 
-		var latestEvent message.Event
-		var prevBootTime int64
+		event, found, err := lastSessionFinder(events, currentEvent, validator, fatalValidator)
 
-		for _, event := range events {
+		if err != nil {
+			return event, err
```

Comment:
Do we want to return the event given if there was an error? Should this be `message.Event{}`?  In `lastSessionFinder()`, when the fatal validation fails, the event it fails on is returned.

Reply:
I return the event it fails on so that whatever uses this could log the error and the event details-is that confusing?

Reply:
Kind of, since the goal of the function is to find an event and you returned one (albeit with an error).  I think anything error related should be returned as a part of the error itself, so it's less confusing and we don't provide an event that a caller could try to use for something.

I'm also okay with creating a more detailed error as a separate pr, and then either leaving things as they are for now or returning an empty error for now.

Reply:
Good point, I'll return an empty event here and I've made an issue to create a more detailed error.